### ISSUE TITLE
code error

### ISSUE THREAD

**Issue by chenhch8** [UCLA-StarAI/Semantic-Loss#1]
Hello, thanks for your releasing code.
Recently, I learned your implemention of semi-supervised classification and found there was an error in semantic.py. The shape of label_examples in your code is (32, 32), which I thought should be (32, 1). I tried to correct the error by adding `keep_dims=True` in `tf.reduce_max` function but found that the  training result was wrong, just as the folloing picture showing:
![1](https://user-images.githubusercontent.com/15072556/46622525-8cd58280-cb5d-11e8-9a6b-c465784a59b4.jpg)
Would you mind telling how to correct it ?
Look forward for your reply!

**Comment by YitaoLiang** [comment]
Hi,
  Would you mind providing more details about this issue. Which dataset were you running on? And can you classify what do you mean the shape of label_examples? Are you referring to the images, whose shape should be (batch_size, 28*28) or labels which should be (batch_size, 10). 
  If you don't mind, you can also zip the files you are using with the changes you made and send them to me at yliang@cs.ucla.edu. It would help me debug the issue.

**Comment by chenhch8** [comment]
Thanks for your reply!
I have sent the zip files to your mail.
